pkg/controller/appdefinition/networkpolicy.go

```diff
@@ -0,0 +1,224 @@
+				} else {
+					resp.Objects(buildNetPolForOtherPublishedPort(
+						fmt.Sprintf("%s-%s-sidecar-%s-%s", strings.ToLower(appName), strings.ToLower(containerName), strings.ToLower(sidecarName), strconv.Itoa(int(port.Port))),
+						podNamespace, podCIDR, containerName, port.Port))
+				}
+			}
+		}
+	}
+}
+
+func buildNetPolForHTTPPublishedPort(name, namespace, ingressNamespace, containerName string, port int32) *networkingv1.NetworkPolicy {
+	var namespaceSelector metav1.LabelSelector
+	if ingressNamespace != "" {
+		namespaceSelector = metav1.LabelSelector{
+			MatchLabels: map[string]string{
+				"kubernetes.io/metadata.name": ingressNamespace,
+			},
+		}
+	}
+
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					labels.AcornContainerName: containerName,
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{{
+				From: []networkingv1.NetworkPolicyPeer{{
+					NamespaceSelector: &namespaceSelector,
+				}},
+				Ports: []networkingv1.NetworkPolicyPort{{
+					Port: &intstr.IntOrString{
+						IntVal: port,
+					}},
+				}},
+			},
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		},
+	}
+}
+
+// For now, we lock down published TCP/UDP ports by allowing access from all pods in kube-system
+// and all IP addresses that aren't part of the pod CIDR.
+// This blocks traffic coming from pods from other projects (since their IPs are in the pod CIDR),
+// but it allows traffic coming from klipper pods in kube-system (which might be doing the load balancing),
+// and from nodes or load balancers that are from a cloud provider.
+func buildNetPolForOtherPublishedPort(name, namespace, podCIDR, containerName string, port int32) *networkingv1.NetworkPolicy {
+	ipBlock := networkingv1.IPBlock{
+		CIDR: "0.0.0.0/0",
+	}
+	if podCIDR != "" {
+		ipBlock.Except = []string{podCIDR}
+	}
+
+	return &networkingv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: networkingv1.NetworkPolicySpec{
+			PodSelector: metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					labels.AcornContainerName: containerName,
+				},
+			},
+			Ingress: []networkingv1.NetworkPolicyIngressRule{{
+				From: []networkingv1.NetworkPolicyPeer{
+					{
+						IPBlock: &ipBlock,
+					},
+					{
+						NamespaceSelector: &metav1.LabelSelector{
+							MatchLabels: map[string]string{
+								"kubernetes.io/metadata.name": "kube-system",
+							},
+						},
+					},
+				},
+				Ports: []networkingv1.NetworkPolicyPort{{
+					Port: &intstr.IntOrString{
+						IntVal: port,
+					}},
+				}},
+			},
+			PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
+		},
+	}
+}
+
+func validateCIDR(cidr string) error {
```

Comment:
Use `net.ParseCIDR` from the standard library instead: https://pkg.go.dev/net#ParseCIDR

Reply:
Dangit! I went looking for something like this but didn't find it so thought "eh I'll make my own". Thankfully I didn't waste too much time...